### ISSUE TITLE
Continuous Integration

### ISSUE THREAD

**Issue by barbieri** [profusion/sgqlc#29]
Setup and integrate a CI system such as travis/circle/jenkins to verify the quality (flake8 + nose tests)

**Comment by barbieri** [comment]
cc: @metachris  if you want to take it

**Comment by metachris** [comment]
Am happy to!

You'd need to allow Travis access to this repository with the first 3 steps of this tutorial: https://docs.travis-ci.com/user/tutorial/#to-get-started-with-travis-ci

* Go to Travis-ci.com and [Sign up with GitHub](https://travis-ci.com/signin).
* Accept the Authorization of Travis CI. You’ll be redirected to GitHub.
* Click the green Activate button, and select the repositories you want to use with Travis CI.

I'll make a PR with a `.travis.yml` file that runs all the tests for every PR.

**Comment by metachris** [comment]
And perhaps you could do the same steps with https://coveralls.io -- we can measure the coverage in the Travis Job and have coveralls leave a nice coverage comment on each PR.

**Comment by barbieri** [comment]
done the travis bit, will do the coveralls later. the travis build runs pre-push, that already checks for coverage, it would be a matter of publish the output

**Comment by barbieri** [comment]
tried to do the coveralls, but the python package seems to read the `.coverage` which varies in format (sqlite or a private/json) depending on `--with-xml` or `--with-html` being executed (at least it seems).

so it's failing on travis, but works on my machine :-/

**Comment by barbieri** [comment]
these are currently set in `setup.cfg` and I'd like to keep html enabled by default.

**Comment by metachris** [comment]
I can't see it on Travis: https://travis-ci.org/profusion/sgqlc

**Comment by metachris** [comment]
Travis runs successfully for me on current master: https://travis-ci.org/metachris/sgqlc/jobs/508354775

**Comment by barbieri** [comment]
what?! It vanished? 👀 will do again

**Comment by barbieri** [comment]
damn, i did the setup in their .com site, seems different instances?

**Comment by metachris** [comment]
Hm, I think .com is the paid version and .com the free one for open-source projects. Definitely should use .org for sgqlc :)

**Comment by barbieri** [comment]
yeah, we now have both... they both offer free versions for open source... I thought they were the same, just different domains and one you could pay to get the private projects
